qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -0,0 +1,557 @@
+            StabilizerState: the tensor product operator self ⊗ other.
+
+        Raises:
+            QiskitError: if other is not a StabilizerState.
+        """
+        if not isinstance(other, StabilizerState):
+            other = StabilizerState(other)
+        return StabilizerState((self.data).tensor(other.data))
+
+    def expand(self, other):
+        """Return the tensor product stabilzier state other ⊗ self.
+
+        Args:
+            other (StabilizerState): a stabilizer state object.
+
+        Returns:
+            StabilizerState: the tensor product operator other ⊗ self.
+
+        Raises:
+            QiskitError: if other is not a StabilizerState.
+        """
+        if not isinstance(other, StabilizerState):
+            other = StabilizerState(other)
+        return StabilizerState((self.data).expand(other.data))
+
+    def evolve(self, other, qargs=None):
+        """Evolve a stabilizer state by a Clifford operator.
+
+        Args:
+            other (Clifford or QuantumCircuit or qiskit.circuit.Instruction):
+                The Clifford operator to evolve by.
+            qargs (list): a list of stabilizer subsystem positions to apply the operator on.
+
+        Returns:
+            StabilizerState: the output stabilizer state.
+
+        Raises:
+            QiskitError: if other is not a StabilizerState.
+            QiskitError: if the operator dimension does not match the
+                         specified StabilizerState subsystem dimensions.
+        """
+        if qargs is None:
+            qargs = getattr(other, 'qargs', None)
+
+        if not isinstance(other, StabilizerState):
+            other = StabilizerState(other)
+        return StabilizerState((self.data).compose(other.data, qargs))
+
+    def expectation_value(self, oper, qargs=None):
+        """Compute the expectation value of an operator.
+
+        Args:
+            oper (BaseOperator): an operator to evaluate expval.
+            qargs (None or list): subsystems to apply the operator on.
+
+        Returns:
+            complex: the expectation value (only 0 or 1 or -1).
+        """
+        num_qubits = self.data.num_qubits
+        if qargs is None:
+            qubits = range(num_qubits)
+        else:
+            qubits = qargs
+
+        # Construct Pauli on num_qubits
+        pauli = Pauli(num_qubits * 'I')
+        phase = 0
+
+        for pos, qubit in enumerate(qubits):
+            pauli_pos = (oper.to_label())[len(oper) - 1 - pos]
+            if pauli_pos == 'X':
+                pauli.x[qubit] = 1
+            elif pauli_pos == 'Y':
+                pauli.x[qubit] = 1
+                pauli.z[qubit] = 1
+                phase += 1
+            elif pauli_pos == 'Z':
+                pauli.z[qubit] = 1
+            else:
+                pass
+
+        # Check if there is a stabilizer that anti-commutes with an odd number of qubits
+        # If so the expectation value is 0
+        for p in range(num_qubits):
```

Comment:
This is a somewhat non-pythonic way of doing this (and one can use numpy arrays to even simplify it further), but it's only a minor comment and I don't think change is really needed right now.

Reply:
Thanks for the comment. I replaced the for loops by `np.array`.